src/main/default/lwc/lookup/lookup.js

```diff
@@ -48,13 +54,14 @@ export default class Lookup extends LightningElement {
 
     @api
     setSearchResults(results) {
+        this._searchResultsCount = results.length;
         // Reset the spinner
         this.loading = false;
         // Clone results before modifying them to avoid Locker restriction
         const resultsLocal = JSON.parse(JSON.stringify(results));
         // Format results
         const regex = new RegExp(`(${this._searchTerm})`, 'gi');
-        this._searchResults = resultsLocal.map((result) => {
+        this._searchResults = resultsLocal.splice(0, this.resultsToDisplay).map((result) => {
```

Comment:
do this in an extra line, so it is easier to merge and see.
Maybe also put the 2 custom lines at the same place, they are highly related to each other and this makes the custom changes more maintainable